sources/kms/key.go

```diff
@@ -0,0 +1,102 @@
+package kms
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go-v2/service/kms/types"
+
+	"github.com/aws/aws-sdk-go-v2/service/kms"
+	"github.com/overmindtech/aws-source/sources"
+	"github.com/overmindtech/sdp-go"
+)
+
+type kmsClient interface {
+	DescribeKey(ctx context.Context, params *kms.DescribeKeyInput, optFns ...func(*kms.Options)) (*kms.DescribeKeyOutput, error)
+	ListKeys(context.Context, *kms.ListKeysInput, ...func(*kms.Options)) (*kms.ListKeysOutput, error)
+	ListResourceTags(context.Context, *kms.ListResourceTagsInput, ...func(*kms.Options)) (*kms.ListResourceTagsOutput, error)
+}
+
+func getFunc(ctx context.Context, client kmsClient, scope string, input *kms.DescribeKeyInput) (*sdp.Item, error) {
+	output, err := client.DescribeKey(ctx, input)
+	if err != nil {
+		return nil, err
+	}
+
+	if output.KeyMetadata == nil {
+		return nil, &sdp.QueryError{
+			ErrorType:   sdp.QueryError_NOTFOUND,
+			ErrorString: "describe key response was nil",
+		}
+	}
+
+	attributes, err := sources.ToAttributesCase(output.KeyMetadata)
+	if err != nil {
+		return nil, err
+	}
+
+	resourceTags, err := tags(ctx, client, *input.KeyId)
+	if err != nil {
+		return nil, &sdp.QueryError{
+			ErrorType:   sdp.QueryError_NOTFOUND,
+			ErrorString: err.Error(),
+		}
+	}
+
+	item := &sdp.Item{
+		Type:            "kms-key",
+		UniqueAttribute: "keyId",
+		Attributes:      attributes,
+		Scope:           scope,
+		Tags:            resourceTags,
+	}
+
+	switch output.KeyMetadata.KeyState {
+	case types.KeyStateEnabled:
+		item.Health = sdp.Health_HEALTH_OK.Enum()
+	case types.KeyStateDisabled:
+		item.Health = sdp.Health_HEALTH_ERROR.Enum()
```

Comment:
```suggestion
		item.Health = sdp.Health_HEALTH_UNKNOWN.Enum()
```

I don't think disabled is necessarily bad, let's not mark it as unhealthy